python/ray/train/_internal/storage.py

```diff
@@ -30,6 +30,9 @@
 from ray.tune.syncer import Syncer, SyncConfig, _BackgroundSyncer
 from ray.tune.result import _get_defaults_results_dir
 
+if TYPE_CHECKING:
+    from ray.train._checkpoint import Checkpoint
```

Comment:
This circular dependency is:
* `ray.train._internal.storage` ->`ray.train._checkpoint` -> `ray.train._internal.storage._download_fs_path`

We can solve this by moving the filesystem utils to a different file.